File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        int i=0, j=0, s=0, ms=Integer.MIN_VALUE;
        while(j<nums.length) {
            s+=nums[j];
            if(j-i+1==k) {
                ms = Math.max(s, ms);
                s-=nums[i];
                i++;
            }
            j++;
        }
        return (double)ms/k;
    }
}

File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        window_sum = sum(nums[:k])
        max_avg = window_sum/k

        for right in range(k, len(nums)):
            window_sum += nums[right]
            window_sum -= nums[right-k]
            max_avg = max(max_avg, window_sum/k)

        return max_avg